File: leetcode/AIcoding/2.Gemini-2.0/javascript_solutions/2813-maximum-elegance-of-a-k-length-subsequence.js
var findMaximumElegance = function(items, k) {
    items.sort((a, b) => b[0] - a[0]);
    let sum = 0;
    let uniqueCategories = new Set();
    let extraProfits = [];

    for (let i = 0; i < k; i++) {
        sum += items[i][0];
        if (!uniqueCategories.has(items[i][1])) {
            uniqueCategories.add(items[i][1]);
        } else {
            extraProfits.push(items[i][0]);
        }
    }

    let maxElegance = sum + uniqueCategories.size * uniqueCategories.size;

    for (let i = k; i < items.length; i++) {
        if (!uniqueCategories.has(items[i][1])) {
            if (extraProfits.length > 0) {
                let minProfit = extraProfits.shift();
                sum -= minProfit;
                sum += items[i][0];
                uniqueCategories.add(items[i][1]);
                maxElegance = Math.max(maxElegance, sum + uniqueCategories.size * uniqueCategories.size);
            }
        }
    }

    return maxElegance;
};

File: leetcode/AIcoding/3.DeepSeek-v3/erlang_solutions/2813-maximum-elegance-of-a-k-length-subsequence.erl
-module(solution).
-export([find_maximum_elegance/2]).

find_maximum_elegance(Items, K) ->
    {Unique, Duplicates} = lists:partition(fun({_, C}) -> C =:= 1 end, Items),
    Sorted = lists:sort(fun({P1, _}, {P2, _}) -> P1 > P2 end, Items),
    {Selected, Rest} = lists:split(K, Sorted),
    UniqueSelected = lists:filter(fun({_, C}) -> C =:= 1 end, Selected),
    Profit = lists:sum([P || {P, _} <- Selected]),
    Distinct = length(lists:usort([C || {_, C} <- Selected])),
    MaxElegance = Profit + Distinct * Distinct,
    case length(UniqueSelected) =:= 0 of
        true -> MaxElegance;
        false ->
            {MinUniqueProfit, MinUniqueCategory} = lists:min([{P, C} || {P, C} <- UniqueSelected]),
            PotentialReplacements = lists:filter(fun({P, C}) -> 
                not lists:member(C, [Cat || {_, Cat} <- Selected]) andalso C =/= MinUniqueCategory
            end, Rest),
            case PotentialReplacements of
                [] -> MaxElegance;
                _ ->
                    {MaxP, MaxC} = lists:max([{P, C} || {P, C} <- PotentialReplacements]),
                    NewProfit = Profit - MinUniqueProfit + MaxP,
                    NewDistinct = Distinct + 1,
                    NewElegance = NewProfit + NewDistinct * NewDistinct,
                    max(MaxElegance, NewElegance)
            end
    end.